### ISSUE TITLE
[Docs] Multitask classification SVGP

### ISSUE THREAD

**Issue by srn284** [cornellius-gp/gpytorch#1396]
I understand there is a SVGP multi _class_ classification, but I believe there is no multitask classification (multiclass preferably). 

E.g. LCM with a multiclass wrap with softmax likelihood on each task prediction? Perhaps the LCM kernel might have to be redefined for this?

Something potentially here in issue [772](https://github.com/cornellius-gp/gpytorch/issues/772) to work with kindly given at the bottom of the page by @jacobrgardner. I see there's a GP layer (although a exactGP) and an NN layer after. Perhaps the GP can be swapped out here for an LCM and then a nn.softmax activation for multiclass on all the LCM prediction head or at least sigmoid? Not sure how to put in the likelihood though.

Many apologies if this is a duplicate post but I couldn't find any tutorials/examples on this. Thank you very much for helping with this.

**Comment by gpleiss** [comment]
If I understand correctly, you want to have a multi-output Gaussian process, with a BernoulliLikelihood on each output dimension? I think this could be accomplished by following the multitask regression SVGP example and replacing the likelihood with the following:

```python
import torch
import gpytorch

class MultitaskBernoulliLikelihood(gpytorch.likelihoods.Likelihood):
    def forward(self, function_samples, **kwargs):
        output_probs = torch.distributions.Normal(0, 1).cdf(function_samples)
        return torch.distributions.Independent(torch.distributions.Bernoulli(probs=output_probs), 1)
```

**Comment by srn284** [comment]
oh yeah, that would be great! This would go along the lines of sigmoidal and that would work indeed. 

To extend this q, how do we do multitask multiclass? I might be able to mess with this MultitaskBernoulliLikelihood class to make it categorical instead of bernoulli. Would you recommend that?

Thank you very much @gpleiss for your time!

**Comment by gpleiss** [comment]
To do multitask/multiclass, you'd do something similar. If you have `c` classes and `t` tasks - the output of your GP model should be a `MultitaskMultivariateNormal` with `c * t` tasks. You would then do something like:

```python
class MultitaskCategoricalLikelihood(gpytorch.likelihoods.Likelihood):
    def forward(self, function_samples, **kwargs):
        function_samples = function_samples.view(*function_samples.shape[:-1], self.t, self.c)  # self.c and self.t should be defined in the constructor
        return torch.distributions.Independent(torch.distributions.Categorical(logits= function_samples), 1)
```

**Comment by mochar** [comment]
> If I understand correctly, you want to have a multi-output Gaussian process, with a BernoulliLikelihood on each output dimension? I think this could be accomplished by following the multitask regression SVGP example and replacing the likelihood with the following:
> 
> ```python
> import torch
> import gpytorch
> 
> class MultitaskBernoulliLikelihood(gpytorch.likelihoods.Likelihood):
>     def forward(self, function_samples, **kwargs):
>         output_probs = torch.distributions.Normal(0, 1).cdf(function_samples)
>         return torch.distributions.Independent(torch.distributions.Bernoulli(probs=output_probs), 1)
> ```

This only worked for me after changing the second parameter of `torch.distributions.Independent` from 1 to 3.
